### PR TITLE
Display test suites in sorted order

### PR DESCRIPTION
**Pull request by eric-gilbertson**
Enhancements to the test runner page:
  * order tests by suite name
  * condense the test failure message so it is easier to read/grok
  * clicking on error summary link toggles display of extended error message
  * increase width of the error display column
condensed_error
expanded_error

### CODE REVIEW

File: testrunner/app/controllers/testrunner.go [hunk 1]
```diff
@@ -191,7 +192,25 @@ func describeSuite(testSuite interface{}) TestSuiteDesc {
 
 // errorSummary gets an error and returns its summary in human readable format.
 func errorSummary(err *revel.Error) (message string) {
-	message = fmt.Sprintf("%4sStatus: %s\n%4sIn %s", "", err.Description, "", err.Path)
+	EXPECTED_PREFIX := "(expected)"
```
Comment (anonx) [inline testrunner/app/controllers/testrunner.go]: Code of `testrunner` has already been `lint`ed. Could you, please, follow Golang's naming conventions. Read more about them here: https://github.com/golang/go/wiki/CodeReviewComments

File: testrunner/app/controllers/testrunner.go [hunk 2]
```diff
@@ -191,7 +192,25 @@ func describeSuite(testSuite interface{}) TestSuiteDesc {
...
+	slashIdx := strings.LastIndex(errFile, "/")
+	if slashIdx > 0 {
+		errFile = errFile[slashIdx+1 : len(errFile)]
+	}
+
+	message = fmt.Sprintf("%s %s#%d", errDesc, errFile, err.Line)
+	return
 
 	// If line of error isn't known return the message as is.
 	if err.Line == 0 {
```
Comment (anonx) [inline testrunner/app/controllers/testrunner.go]: This line of code is not reachable.

File: testrunner/app/controllers/testrunner.go [hunk 3]
```diff
@@ -234,6 +253,15 @@ func formatResponse(t testing.TestSuite) map[string]string {
 	}
 }
 
+//functions needed to sort the testsuites by name.
+type SortBySuiteName []interface{}
```
Comment (anonx) [inline testrunner/app/controllers/testrunner.go]: I'd like all public types and functions to have comments in the format:
```go
// TypeName ...
```

**Comment by brendensoares** [comment]
1. Per my comments in #12 can we enlarge the run buttons? Perhaps just add some padding to them so they're easier to click?
2. If we could vertically center the "Run All Tests" buttons that'd be nice.
3. Can we label suites as such? e.g. "**Suite** LoginCtrlTest"
4. Thoughts on collapsing suites by default and auto expanding if there is a failure?

**Comment by eric-gilbertson** [comment]
I have taken care of the requested items. For item #4, I modified the test results handler to automatically open the panel for the failed test suites. I tried #3, but I felt that it the extra text made it more difficult to quickly grok the test name. Take a look at the two attachments (with and without the "Suite" text) and let me know if you still want it in. Given that this page is used exclusively by developers,  I think that both the functionality and aesthetics of it are best served by a condensed layout that allows one to quickly run tests and view the results with a minimum of scrolling. 

The one additional item I was considering was replacing the "Test Runner" heading text with the "revelhat" image in order to better convey that this page is part of the Revel infrastructure.  Please advise whether you want this change and if so whether I should do this by copying the image into 'modules/testrunner/public/images (currently I set the img src to the github   page version).

Regards,

Eric G.

PS
I presume that you have noticed that the you can get the full error display by clicking on the error summary link. 
testrunner_nosuite
testrunner_suite

**Comment by eric-gilbertson** [comment]
Note, the changes discussed in the previous comment have not been pushed. I wanted to get your okay before committing them.

**Comment by eric-gilbertson** [comment]
This commit contains the changes described in #12 with the exception of the Revel logo in the header. Changes include:
   * expand closed display panels when executing "test all" if suite contains an error
   * increase size of the Run buttons
   * vertically center the Run All Tests button

The screen shots for this commit are the same as shown in #12 with the exception of the logo.

**Comment by brendensoares** [comment]
I think this looks great now. Let's merge and go from there!

**Pull request merged**